### DISCUSSION TITLE
Download wix package fail...help

### DISCUSSION THREAD

**Discussion by peterroe** [tauri-apps/tauri#3770] [Q&A] [answered]
When I run:

```shell
$ npx vue-cli-service tauri:build
```

The error happend:

```shell
Compiling app v0.1.0 (D:\VscodeItems\tauri-app\src-tauri)
    Finished release [optimized] target(s) in 22.91s
info: Verifying wix package
info: Downloading https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311-binaries.zip
(node:10460) UnhandledPromiseRejectionWarning: Error: failed to bundle project: `Tls Error: The remote host forcibly closed an existing connection. (os error 10054)`: Tls Error: An existing connection was forcibly closed by the remote host. (os error 10054)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:10460) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:10460) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I dont't know what should I do... help  :(

my node env: 

```shell
$ node v14+
```

I try to upgrade my node version to `v16/v17`，it prompts another error:

```shell
Compiling app v0.1.0 (D:\VscodeItems\tauri-app\src-tauri)
     Finished release [optimized] target(s) in 21.85s
info: Verifying wix package
info: Downloading https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311-binaries.zip
node:internal/process/promises:279
             triggerUncaughtException(err, true /* fromPromise */);
             ^

[Error: failed to bundle project: `Io Error: No such host known. (os error 11001)`: Io Error: No such host known. (os error 11001)] {
   code: 'GenericFailure'
}
```

**Comment by FabianLars** [comment, marked as answer]
It's probably blocked by a firewall or something. Can you try downloading [it](https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311-binaries.zip) manually and unpack it into the src-tauri dir so that the project structure looks similar to this:
```
./
  src/
  src-tauri/
    WixTools/
      many files, for example "candle.exe"
```

**Reply by peterroe** [reply]
Thanks for helping me on the project! I really appreciate it! :) 👍

**Reply by 7086cmd** [reply]
Sorry, this issue shows again in my project. It also wanna download this compressed file after I unziped into the current folder that you said.

**Reply by peterroe** [reply]
Are you sure? This shouldn't happen if you do as I said.

**Reply by FabianLars** [reply]
Actually the location of the WixTools directory changed. It is now in `C:\Users\your-username\AppData\Local\tauri\WixTools`

**Reply by liudonghua123** [reply]
Yes, the location is changed, you should extract [`wix311-binaries.zip`](https://github.com/wixtoolset/wix3/releases/download/wix3112rtm/wix311-binaries.zip) to `%LOCALAPPDATA%\tauri\WixTools`, not put `wix311-binaries.zip` in `%LOCALAPPDATA%\tauri\WixTools` or you will got `Warn WixTools directory is missing some files. Recreating it.` warnings.

See https://github.com/tauri-apps/tauri/blob/b7277357b960521b04b4d9153bd402953cd483df/tooling/bundler/src/bundle/windows/msi.rs#L28C1-L44

**Reply by liudonghua123** [reply]
And you should also do the similar things for nsis. Extract [nsis-3.zip](https://github.com/tauri-apps/binary-releases/releases/download/nsis-3/nsis-3.zip) to `%LOCALAPPDATA%\tauri\NSIS`, [NSIS-ApplicationID.zip/ReleaseUnicode/ApplicationID.dll](https://github.com/tauri-apps/binary-releases/releases/download/nsis-plugins-v0/NSIS-ApplicationID.zip) to `%LOCALAPPDATA%\tauri\NSIS\Plugins\x86-unicode` and [nsis_tauri_utils.dll](https://github.com/tauri-apps/nsis-tauri-utils/releases/download/nsis_tauri_utils-v0.1.1/nsis_tauri_utils.dll) to `%LOCALAPPDATA%\tauri\NSIS\Plugins\x86-unicode`.

See https://github.com/tauri-apps/tauri/blob/b7277357b960521b04b4d9153bd402953cd483df/tooling/bundler/src/bundle/windows/nsis.rs#L68-L84

and https://github.com/tauri-apps/tauri/blob/b7277357b960521b04b4d9153bd402953cd483df/tooling/bundler/src/bundle/windows/nsis.rs#L87-L122

**Comment by duyl328** [comment]
I also encountered the same error, and my computer could not connect directly to the Internet. I tried to decompress the Wix compressed package I downloaded manually under :

`%LOCALAPPDATA%\tauri\WixTools`, 

but the error still persisted. After querying the source code, I found that `WixTools` The folder name has been changed to: 

`WixTools314`.

The relevant code is as follows:

https://github.com/tauri-apps/tauri/blob/b37741da6a2d3dad71490c910a64eeedda2ba9ca/crates/tauri-bundler/src/bundle/windows/msi/mod.rs#L64

Hope it helps those who encounter the same problem : )
